simba/report.py

```diff
@@ -360,3 +381,259 @@ def write_csv(data: Iterable, file_path, propagate_errors=False):
+        bins[schedule.rotations[rot].vehicle_type][position] += 1
+
+    # plot
+    fig, ax = plt.subplots()
+    bar_bottom = [0] * num_bins
+    for v_type in schedule.vehicle_types:
+        ax.bar(labels, bins[v_type], width=0.9, label=v_type, bottom=bar_bottom)
+        for i in range(num_bins):
+            bar_bottom[i] += bins[v_type][i]
+    ax.set_xlabel('Energy consumption [kWh]')
+    plt.xticks(rotation=30)
+    ax.set_ylabel('Number of rotations')
+    ax.yaxis.get_major_locator().set_params(integer=True)
+    ax.yaxis.grid(True)
+    ax.set_title('Distribution of bus types over rotation energy consumption')
+    ax.legend()
+    plt.savefig(extended_plots_path / "distribution_bustypes_consumption_rotations")
+    plt.close()
+
+
+def plot_charge_type_distribution(extended_plots_path, scenario, schedule):
+    """Plots the number of rotations of each charging type in a bar chart.
+
+    :param extended_plots_path: directory to save plot to
+    :type extended_plots_path: Path
+    :param scenario: Scenario for with to generate timeseries.
+    :type scenario: spice_ev.Scenario
+    :param schedule: Driving schedule for the simulation. schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    """
+    # count charging types (also with regard to negative rotations)
+    charging_types = {'oppb': 0, 'oppb_neg': 0, 'depb': 0, 'depb_neg': 0}
+    negative_rotations = schedule.get_negative_rotations(scenario)
+    for rot in schedule.rotations:
+        ct = schedule.rotations[rot].charging_type
+        if rot in negative_rotations:
+            ct += '_neg'
+        try:
+            charging_types[ct] += 1
+        except KeyError:
+            logging.error(f"Rotation {rot}: unknown charging type: '{ct}'")
+
+    # plot
+    fig, ax = plt.subplots()
+    bars1 = ax.bar(
+        ["Opportunity", "Depot"],
+        [charging_types["oppb"], charging_types["depb"]],
+        color=["#66CDAA", "#66CDAA"],
+    )
+    bars2 = ax.bar(
+        ["Opportunity", "Depot"],
+        [charging_types["oppb_neg"], charging_types["depb_neg"]],
+        color=["#6495ED", "#6495ED"],
+        bottom=[charging_types["oppb"], charging_types["depb"]],
+    )
+    # create labels with counts
+    # create empty labels for empty bins
+    labels = [
+        [
+            f"{ct}{suffix}: {charging_types[f'{ct}{suffix}']}"
+            if charging_types[f'{ct}{suffix}'] > 0 else ""
+            for ct in ['oppb', 'depb']
+        ] for suffix in ['', '_neg']
+    ]
+    ax.bar_label(bars1, labels=labels[0], label_type='center')  # oppb, depb
+    ax.bar_label(bars2, labels=labels[1], label_type='center')  # oppb_neg, depb_neg
+
+    ax.set_xlabel("Charging type")
+    ax.set_ylabel("Number of rotations")
+    ax.yaxis.grid(True)
+    ax.yaxis.get_major_locator().set_params(integer=True)
+    ax.legend(["successful rotations", "negative rotations"])
+    ax.set_title("Distribution of opportunity and depot charging")
```

Comment:
"Negative rotations per charging type"?

Reply:
I would go with "Feasibility of rotations per charging type"